packages/itwinui-css/src/input/input-flex-container.scss

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+// See LICENSE.md in the project root for license terms and full copyright notice.
+@mixin iui-input-flex-container {
+  --_iui-input-background-color: var(--iui-color-background);
+  --_iui-input-border-color: var(--iui-color-border-foreground);
+  --_iui-input-text-color: var(--iui-color-text);
+
+  @include iui-reset;
+  @include iui-focus($offset: -2px, $thickness: 2px);
+  width: 100%;
+  font-family: inherit;
+  font-size: var(--iui-font-size-1);
+  font-weight: var(--iui-font-weight-normal);
+  line-height: var(--iui-size-l);
+  border-radius: var(--iui-border-radius-1);
+  appearance: none;
+  height: var(--_iui-input-min-height);
+  padding-block: var(--_iui-input-padding-block);
+  padding-inline: var(--iui-size-s);
+  color: var(--_iui-input-text-color);
+  background-color: var(--_iui-input-background-color);
+  border: 1px solid var(--_iui-input-border-color);
+  transition: border-color var(--iui-duration-1) ease-out;
+  @include iui-input-size;
+
+  right: unset;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  padding: 0;
+  margin: 0;
+
+  > input {
+    padding-left: var(--iui-size-s);
+    width: 100%; // Take all available space
+    color: var(--iui-color-text);
+    font-family: inherit;
+    font-size: var(--iui-font-size-1);
+    font-weight: var(--iui-font-weight-normal);
+    line-height: var(--iui-size-l);
+
+    &::placeholder {
+      --_iui-input-text-color: var(--iui-color-text-disabled);
+      user-select: none;
+      color: var(--_iui-input-text-color);
+    }
+  }
+
+  &:autofill,
+  > input:autofill {
+    --_iui-input-background-color: var(--iui-color-background-accent-muted);
+    --_iui-input-border-color: var(--iui-color-border-accent);
+    --_iui-input-text-color: var(--iui-color-text-accent);
+  }
+
+  &[disabled],
+  > input[disabled] {
+    --_iui-input-background-color: var(--iui-color-background-disabled);
+    --_iui-input-border-color: var(--iui-color-border-disabled);
+    --_iui-input-with-icon-hover-border-color: var(--iui-color-border-disabled);
+    cursor: not-allowed;
+  }
+
+  > .iui-svg-icon {
+    margin-left: var(--iui-size-xs);
+  }
+}
+
+@mixin iui-expandable-searchbox {
+  width: var(--iui-component-height);
+  transition: width var(--iui-duration-3) ease;
+  overflow: hidden;
+
+  &.iui-animate-left {
+    justify-content: flex-end;
```

Comment:
This is causing a slight shift of icon. Maybe @mayank99 or @FlyersPh9 knows how to fix it?
![image](https://user-images.githubusercontent.com/83585998/217025245-e7791934-6b63-4925-9b1b-d894b638ddcd.png)

Reply:
I'm thinking it's because `.iui-expandable-searchbox` and the nested `.iui-button` are both the same width/height with a border.  If `.iui-expandable-searchbox` is 36x36px with a border that gives `.iui-button` only 34x34px to display (because 2px are used by the border).  I tested by changing `.iui-expandable-searchbox` to a width of 38px which does stop the icon shift.

If you make the border a `::before` I think that will allow you to keep the sizes as is without the icon shift.